urld58fd5e7a9_longrm_weibo_robot_tgz-pJ8-weibo4j_org_json_TestJ8/src/weibo4j/User.java

```diff
@@ -0,0 +1,706 @@
+//                List<User> users = new ArrayList<User>(size);
+//                for (int i = 0; i < size; i++) {
+//                    users.add(new User(res, (Element) list.item(i), weibo));
+//                }
+                
+                //去除掉嵌套的bug
+                NodeList list=doc.getDocumentElement().getChildNodes();
+                List<User> users = new ArrayList<User>(list.getLength());
+                Node node;
+                for(int i=0;i<list.getLength();i++){
+                	node=list.item(i);
+                	if(node.getNodeName().equals("user")){
+                		users.add(new User(res, (Element) node, weibo));
+                	}
+                }
+
+                return users;
+            } catch (WeiboException te) {
+                if (isRootNodeNilClasses(doc)) {
+                    return new ArrayList<User>(0);
+                } else {
+                    throw te;
+                }
+            }
+        }
+    }
+
+    public static UserWapper constructWapperUsers(Response res, Weibo weibo) throws WeiboException {
+		Document doc = res.asDocument();
+		if (isRootNodeNilClasses(doc)) {
+			return new UserWapper(new ArrayList<User>(0), 0, 0);
+		} else {
+			try {
+				ensureRootNodeNameIs("users_list", doc);
+				Element root = doc.getDocumentElement();
+				NodeList user = root.getElementsByTagName("users");
+				int length = user.getLength();
+				if (length == 0) {
+					return new UserWapper(new ArrayList<User>(0), 0, 0);
+				}
+				// 
+				Element listsRoot = (Element) user.item(0);
+                NodeList list=listsRoot.getChildNodes();
+                List<User> users = new ArrayList<User>(list.getLength());
+                Node node;
+                for(int i=0;i<list.getLength();i++){
+                	node=list.item(i);
+                	if(node.getNodeName().equals("user")){
+                		users.add(new User(res, (Element) node, weibo));
+                	}
+                }
+				//
+				long previousCursor = getChildLong("previous_curosr", root);
+				long nextCursor = getChildLong("next_curosr", root);
+				if (nextCursor == -1) { // 兼容不同标签名称
+					nextCursor = getChildLong("nextCurosr", root);
+				}
+				return new UserWapper(users, previousCursor, nextCursor);
+			} catch (WeiboException te) {
+				if (isRootNodeNilClasses(doc)) {
+					return new UserWapper(new ArrayList<User>(0), 0, 0);
+				} else {
+					throw te;
+				}
+			}
+		}
+	}
+
+    public static List<User> constructUsers(Response res) throws WeiboException {
+    	try {
+            JSONArray list = res.asJSONArray();
+            int size = list.length();
+            List<User> users = new ArrayList<User>(size);
+            for (int i = 0; i < size; i++) {
+                users.add(new User(list.getJSONObject(i)));
+            }
+            return users;
+        } catch (JSONException jsone) {
+            throw new WeiboException(jsone);
+        } catch (WeiboException te) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: An exception is caught and re-thrown within the catch clause without any other operations being performed on the exception, making the catch clause unnecessary.

**Fix**: Perform additional operations on the exception that has been caught. You can wrap a standard exception into a custom exception to add additional information or special handling. When you do that, make sure to set the original exception as the cause. Otherwise, you will lose the stack trace and message of the original exception, which will make it difficult to analyze the event that caused the exception.  
[Learn more](https://www.ibm.com/support/pages/best-practice-catching-and-re-throwing-java-exceptions)